tests/irdl/test_declarative_assembly_format.py

```diff
@@ -1748,10 +1748,12 @@ class OneOperandOneResultNestedOp(IRDLOperation):
     %1 = test.one_operand_one_result_nested %0 : i32"""
     )
     with pytest.raises(
-        ParseError,
-        match="Verification error while inferring operation type: ",
+        VerifyException,
+        match="i32 should be of base attribute test.param_one",
```

Comment:
This seems like it conveys the message a bit more directly
```suggestion
        match="unexpected attribute i32, expected test.param_one",
```

Reply:
hmm actually is this just an existing constraint string? In that case I won't recommend PR creep, it might just be worth changing in the future.

Reply:
The key change here is that the exception happens during verification, rather than at parse time